cloudinary-image-management-and-manipulation-in-the-cloud-cdn/php/sync/class-upload-queue.php

```diff
@@ -233,17 +261,41 @@ public function stop_queue() {
 			$this->set_queue( $queue );
 		}
 		$this->running = false;
-	}
 
+		wp_clear_scheduled_hook( 'cloudinary_resume_queue' );
+	}
 
 	/**
-	 * Startthe queue by setting the started flag.
+	 * Start the queue by setting the started flag.
+	 *
+	 * @return void
 	 */
 	public function start_queue() {
 		$queue            = $this->get_queue();
-		$queue['started'] = current_time( 'timestamp' );
+		$queue['started'] = $queue['last_update'] = current_time( 'timestamp' );
 		$this->set_queue( $queue );
 		$this->running = true;
+
+		if ( ! wp_next_scheduled( 'cloudinary_resume_queue' ) ) {
+			wp_schedule_single_event( time() + $this->cron_frequency, 'cloudinary_resume_queue' );
+		}
 	}
 
+	/**
+	 * Maybe resume the queue.
+	 * This is a fallback mechanism to resume the queue when it stops unexpectedly.
+	 *
+	 * @return void
+	 */
+	public function maybe_resume_queue() {
+		$now   = current_time( 'timestamp' );
+		$queue = $this->get_queue();
+
+
+		if ( $now - $queue['last_update'] > $this->cron_start_offset ) {
```

Comment:
@pereirinha - Shouldn't we check to see if the queue is running as well, before starting it again?

Reply:
@DavidCramer Good catch. Thank you.